cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -224,12 +299,111 @@ public void insertOrUpdateAssociation(
 			Association association,
 			AssociationContext associationContext) {
 
-		// TODO
+		Table tableMetadata = provider.getMetaDataCache().get( key.getTable() );
+		Set<String> keyColumnNames = new HashSet<String>();
+		for ( Object columnObject : tableMetadata.getPrimaryKey().getColumns() ) {
+			Column column = (Column) columnObject;
+			keyColumnNames.add( column.getName() );
+		}
+
+		tableMetadata.getPrimaryKey().getColumns();
+
+		List<AssociationOperation> updateOps = new ArrayList<AssociationOperation>(
+				association.getOperations()
+						.size()
+		);
+		List<AssociationOperation> deleteOps = new ArrayList<AssociationOperation>(
+				association.getOperations()
+						.size()
+		);
+
+		for ( AssociationOperation op : association.getOperations() ) {
+			switch ( op.getType() ) {
+				case CLEAR:
+					break;
+				case PUT:
+					updateOps.add( op );
+					break;
+				case REMOVE:
+					deleteOps.add( op );
+					break;
+				default:
+					throw new HibernateException( "AssociationOperation not supported: " + op.getType() );
+			}
+		}
+
+		for ( AssociationOperation op : updateOps ) {
+			Tuple value = op.getValue();
+			List<Object> columnValues = new ArrayList<>();
+			Insert insert = insertInto( quote( key.getTable() ) );
+			for ( String columnName : value.getColumnNames() ) {
+				insert.value( quote( columnName ), QueryBuilder.bindMarker( columnName ) );
+				columnValues.add( value.get( columnName ) );
+			}
+
+			bindAndExecute( columnValues.toArray(), insert );
+		}
+
+		for ( AssociationOperation op : deleteOps ) {
+
+			RowKey value = op.getKey();
+			Delete.Selection deleteSelection = delete();
+			for ( String columnName : op.getKey().getColumnNames() ) {
+				if ( !keyColumnNames.contains( columnName ) ) {
+					deleteSelection.column( quote( columnName ) );
+				}
+			}
+			Delete delete = deleteSelection.from( quote( key.getTable() ) );
+			List<Object> columnValues = new LinkedList<>();
+			for ( String columnName : value.getColumnNames() ) {
+				if ( keyColumnNames.contains( columnName ) ) {
+					delete.where( eq( quote( columnName ), QueryBuilder.bindMarker( columnName ) ) );
+					columnValues.add( value.getColumnValue( columnName ) );
+				}
+			}
+
+			bindAndExecute( columnValues.toArray(), delete );
+		}
 	}
 
 	@Override
 	public void removeAssociation(AssociationKey key, AssociationContext associationContext) {
-		// TODO
+
+		if ( key.getMetadata().isInverse() ) {
+			return;
+		}
+
+		Table tableMetadata = provider.getMetaDataCache().get( key.getTable() );
+		Set<String> keyColumnNames = new HashSet<String>();
```

Comment:
Ah, I think it is `key.getMetadata().getRowKeyColumnNames()` what you are after here.

The OGM key (meta-data) should give you everything needed, if not we may have to add something. But the usage of `Table` makes me a bit wary ;)